tools/yulPhaser/Chromosome.cpp

```diff
@@ -0,0 +1,148 @@
+#include <libyul/optimiser/RedundantAssignEliminator.h>
+#include <libyul/optimiser/Rematerialiser.h>
+#include <libyul/optimiser/SSAReverser.h>
+#include <libyul/optimiser/SSATransform.h>
+#include <libyul/optimiser/StructuralSimplifier.h>
+#include <libyul/optimiser/Suite.h>
+#include <libyul/optimiser/UnusedPruner.h>
+#include <libyul/optimiser/VarDeclInitializer.h>
+#include <libyul/optimiser/VarNameCleaner.h>
+
+#include <sstream>
+
+using namespace std;
+using namespace solidity::yul;
+
+namespace solidity::phaser
+{
+
+Chromosome Chromosome::makeRandom(size_t _length)
+{
+	vector<string> steps;
+	for (size_t i = 0; i < _length; ++i)
+		steps.push_back(randomOptimisationStep());
+
+	return Chromosome(move(steps));
+}
+
+ostream& operator<<(ostream& _stream, Chromosome const& _chromosome)
+{
+	for (auto const& stepName: _chromosome.m_optimisationSteps)
+		_stream << OptimiserSuite::stepNameToAbbreviationMap().at(stepName);
+
+	return _stream;
+}
+
+vector<string> Chromosome::allStepNames()
+{
+	vector<string> stepNames;
+	for (auto const& step: OptimiserSuite::allSteps())
+		stepNames.push_back(step.first);
+
+	return stepNames;
+}
+
+vector<string> Chromosome::allStepNamesExcept(vector<string> const& _excludedStepNames)
+{
+	vector<string> all = allStepNames();
+
+	for (string const& stepName: _excludedStepNames)
+	{
+		// This is O(n*m) but vectors are small and we cache the results anyway. What matters a bit
+		// more is that using vector rather than a set gives us O(1) access to random elements in
+		// other functions.
+		auto matchingStep = find(all.begin(), all.end(), stepName);
+		if (matchingStep != all.end())
+			all.erase(matchingStep);
+	}
+
+	return all;
+}
+
+string const& Chromosome::randomOptimisationStep()
+{
+	static vector<string> stepNames = allStepNamesExcept({
+		// All possible steps, listed and commented-out for easy tweaking.
+		// The uncommented ones are not used (possibly because they fail).
+		//{BlockFlattener::name},
+		//{CommonSubexpressionEliminator::name},
+		//{ConditionalSimplifier::name},
+		//{ConditionalUnsimplifier::name},
+		//{ControlFlowSimplifier::name},
+		//{DeadCodeEliminator::name},
+		//{EquivalentFunctionCombiner::name},
+		//{ExpressionInliner::name},
+		//{ExpressionJoiner::name},
+		//{ExpressionSimplifier::name},
+		//{ExpressionSplitter::name},
+		//{ForLoopConditionIntoBody::name},
+		//{ForLoopConditionOutOfBody::name},
+		//{ForLoopInitRewriter::name},
+		//{FullInliner::name},
+		//{FunctionGrouper::name},
+		//{FunctionHoister::name},
+		//{LiteralRematerialiser::name},
+		//{LoadResolver::name},
+		//{LoopInvariantCodeMotion::name},
+		//{RedundantAssignEliminator::name},
+		//{Rematerialiser::name},
+		//{SSAReverser::name},
+		//{SSATransform::name},
+		//{StructuralSimplifier::name},
+		//{UnusedPruner::name},
+		//{VarDeclInitializer::name},
+
+		// VarNameCleaner is not safe to use here because it leaves name dispenser in an unusable
```

Comment:
We might want to take this step out of this tool altogether, it is just a prettifier.

Reply:
Should I remove it from `OptimizerSuite` in this PR?

Reply:
Yes, please!